### ISSUE TITLE
Linting error when dot inside quoted identifier (Snowflake)

### ISSUE THREAD

**Issue by lwehle** [sqlfluff/sqlfluff#4929]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Hello,

I used a dot inside a quoted identifier and SQLFluff marked it as an error. Somehow it seems to get confused with the additional dot.

### Expected Behaviour

I would expect that SQLFluff ignores everything inside the quotes.

### Observed Behaviour

Got the following linting error:
```
L:   9 | P:   8 | RF01 | 
Reference 'C1."DUMMY.ALIAS"' refers to table/view not
found in the FROM clause or found in ancestor statement.
[references.from]
```

### How to reproduce

```sql
WITH CTE_DUMMY AS (
    SELECT
        1 AS "DUMMY.ALIAS"
        , 2 AS "DUMMY_ALIAS"
    FROM DUAL
)

SELECT
    C1."DUMMY.ALIAS"
    , C1."DUMMY_ALIAS"
FROM CTE_DUMMY AS C1
```

### Dialect

Snowflake

### Version

SQLFluff 2.1.1
sqlfluff-templater-dbt 2.1.1
dbt 1.5.1

### Configuration

[sqlfluff]
templater = dbt
dialect = snowflake

# Comma separated list of rules to exclude, or None
exclude_rules = RF02, AL07, ST07, ST06, ST01, CV02
# Ignore file byte limit which default is set to 20000
large_file_skip_byte_limit = 40000
max_line_length = 120

# Layout configuration
[sqlfluff:layout:type:comma]
line_position = leading

[sqlfluff:indentation]
allow_implicit_indents = True
indented_on_contents = True

### rules configuration
# capitalisation
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

# ambigous
[sqlfluff:rules:ambiguous.column_references]
group_by_and_order_by_style = explicit

# aliasing
[sqlfluff:rules:aliasing.expression]
allow_scalar = False

#layout
[sqlfluff:rules:layout.select_targets]
# Select on new line unless there is only one
wildcard_policy = multiple

# structure
[sqlfluff:rules:structure.subquery]
forbid_subquery_in = both

# references
[sqlfluff:rules:references.special_chars]
quoted_identifiers_policy = none

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
